allennlp/models/crf_tagger.py

```diff
@@ -34,15 +35,34 @@ class CrfTagger(Model):
         Unless you did something unusual, the default value should be what you want.
     feedforward : ``FeedForward``, optional, (default = None).
         An optional feedforward layer to apply after the encoder.
-    dropout:  ``float``, optional (detault=``None``)
-    verbose_metrics : ``bool``, optional (default = False)
-        If true, metrics will be returned per label class in addition
-        to the overall statistics.
+    label_encoding : ``str``, optional (default=``None``)
+        Label encoding to use when calculating span f1 and constraining
+        the CRF at decoding time . Valid options are "BIO", "BIOUL", "IOB1".
+        Required if ``calculate_span_f1`` or ``constrain_crf_decoding`` is true.
     constraint_type : ``str``, optional (default=``None``)
         If provided, the CRF will be constrained at decoding time
-        to produce valid labels based on the specified type (e.g. "BIO", or "BIOUL").
+        to produce valid labels based on the specified type
+        (e.g. "BIO", or "BIOUL").
+
+        .. deprecated:: 0.6.1
+           ``constraint_type`` was deprecated and replaced with
+           ``label_encoding``, ``constrain_crf_decoding``, and
+           ``calculate_span_f1`` in version 0.6.1. It will be removed
+           in version 0.8.
+
     include_start_end_transitions : ``bool``, optional (default=``True``)
         Whether to include start and end transition parameters in the CRF.
+    constrain_crf_decoding : ``bool``, optional (default=``False``)
+        If ``True``, the CRF is constrained at decoding time to
+        produce valid sequences of tags. If this is ``True``, then
+        ``label_encoding`` is required.
+    calculate_span_f1 : ``bool``, optional (default=``False``)
+        Calculate span-level F1 metrics during training. If this is ``True``, then
+        ``label_encoding`` is required.
```

Comment:
I know it makes the code more complicated, but would it be better for the default (for calculate_span_f1 and constrain_crf_decoding) to be `None`, and to interpret that as "true if label_encoding supplied, false otherwise", since it seems like that's probably the behavior you'd want?

Reply:
Hmm, yeah i guess that's a more reasonable default.